alacritty_terminal/src/clipboard.rs

```diff
@@ -24,74 +25,139 @@ use copypasta::wayland_clipboard;
 use copypasta::x11_clipboard::{Primary as X11SelectionClipboard, X11ClipboardContext};
 use copypasta::{ClipboardContext, ClipboardProvider};
 
+use crate::event::{Event, EventListener};
+use crate::util;
+
 pub struct Clipboard {
+    _thread: std::thread::JoinHandle<()>,
+    tx: mpsc::Sender<ClipboardRequest>,
+}
+
+pub struct ClipboardStore {
     clipboard: Box<dyn ClipboardProvider>,
     selection: Option<Box<dyn ClipboardProvider>>,
 }
 
+pub enum ClipboardType {
+    Clipboard,
+    Selection,
+}
+
+enum ClipboardAction {
+    Store(String),
+    Load,
+}
+
+struct ClipboardRequest {
+    action: ClipboardAction,
+    ty: ClipboardType,
+}
+
 impl Clipboard {
     #[cfg(any(target_os = "macos", target_os = "windows"))]
-    pub fn new() -> Self {
-        Self::default()
+    pub fn new<T: EventListener + Send + 'static>(event_proxy: T) -> Self {
+        let clipboard = ClipboardStore {
+            clipboard: Box::new(ClipboardContext::new().unwrap()),
+            selection: None,
+        };
+
+        let (tx, rx) = mpsc::channel();
+        Self {
+            _thread: util::thread::spawn_named("clipboard", move || {
+                clipboard_thread(clipboard, rx, event_proxy);
+            }),
```

Comment:
Why do we permanently spawn a thread for the clipboard instead of spawning one on-demand? That's not really great for system resources to just have a useless thread hanging around for no reason most of the time.